### PR TITLE
addressing issue #129

### PR DESCRIPTION
**Pull request by ejiawustl**
made changes to address all 3 points in issue #129 - updated typehinting, docstring, and return results.

### CODE REVIEW

File: yeastdnnexplorer/__main__.py [hunk 1]
```diff
@@ -365,7 +396,7 @@ def find_interactors_workflow(args: argparse.Namespace) -> None:
         "final_model_avg_r_squared": final_model_avg_r_squared,
     }
 
-    output_path = os.path.join(args.output_dir, f"{args.response_tf}_output.json")
+    output_path = os.path.join(output_dirpath, f"{args.response_tf}_final_output.json")
```
Comment (cmatKhan) [inline yeastdnnexplorer/__main__.py:399]: Why does this include the response_tf in the output name? it is going in `args.output_dirpath/response_tf`. Can't this just be called "final_output.json"?

The point is that it makes it easier to grep out files if they all have the same name

File: yeastdnnexplorer/ml_models/lasso_modeling.py [hunk 2]
```diff
@@ -652,13 +655,18 @@ def get_significant_predictors(
...
         ValueError(f"method {method} not recognized")
 
-    return {
+    result = {
         "sig_coefs": sig_coef_dict,
         "predictors": X,
         "response": y,
         "classes": stratification_classes,
+        "bootstrap_lasso_output": (
+            bootstrap_lasso_output if method == "bootstrap_lassocv" else None
```
Comment (cmatKhan) [inline yeastdnnexplorer/ml_models/lasso_modeling.py:664]: good

File: yeastdnnexplorer/ml_models/lasso_modeling.py [hunk 3]
```diff
@@ -568,10 +568,13 @@ def get_significant_predictors(
         are 'quantile_threshold' from generate_modeling_data() and 'ci_percentile' from
         examine_bootstrap_coefficients()
 
-    :return a dictionary with keys 'sig_coefs', 'response', 'classes' where:
+    :return a dictionary with keys 'sig_coefs', 'response', 'classes', and
+        'bootstrap_lasso_output' where:
         - 'sig_coefs' is a dictionary of significant coefficients
         - 'response' is the response variable
         - 'classes' is the stratification classes for the data
+        - 'bootstrap_lasso_output' is the bootstrapping output for intermediate model
```
Comment (cmatKhan) [inline yeastdnnexplorer/ml_models/lasso_modeling.py:576]: note that it will be none if method is lassocv_ols (or wahtever the other method is)

**Review by cmatKhan — CHANGES_REQUESTED** [review]

File: yeastdnnexplorer/ml_models/lasso_modeling.py (hunk 3 above)
Comment (ejiawustl) [reply yeastdnnexplorer/ml_models/lasso_modeling.py:576]: Added in most recent push

File: yeastdnnexplorer/__main__.py (hunk 1 above)
Comment (ejiawustl) [reply yeastdnnexplorer/__main__.py]: Changed in most recent push

**Review by cmatKhan — APPROVED** [review]

**Comment by cmatKhan** [comment]
closes #129

@ejiawustl  -- 'closes` is a keyword in git. these are useful:

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

**Pull request merged**